spacesuit/src/bit_range.rs

```diff
@@ -0,0 +1,31 @@
+/// Represents a usize with value in the range [0,64]
+#[derive(Copy, Clone, Debug)]
+pub struct BitRange(usize);
+
+impl BitRange {
+    /// Returns Some(BitRange) if `n` is ≤ 64.
+    /// Otherwise returns None.
+    pub fn new(n: usize) -> Option<Self> {
+        if n > 64 {
+            None
+        } else {
+            Some(BitRange(n))
+        }
+    }
+
+    /// Returns 64-bit range
+    pub fn max() -> Self {
+        BitRange(64)
+    }
+
+    /// Converts BitRange to a `u8`
+    pub fn to_u8(self) -> u8 {
```

Comment:
why not `Into<u8>` like with `Into<usize>`?

Reply:
changed. It could not determine the type with .into when there was more than one into impl for some reason. I used an extra line to help the compiler : https://github.com/interstellar/slingshot/pull/184/commits/145ebff13dde8d2aad74a626bb87a7352f59aae7#diff-9cf510fc0d33929cbc49a13f477fe8f7R259

Let me know if there is a better solution

Reply:
I'm curious what was the error? It seems to me the `.push` is clearly expecting `u8`, and the Range op has `BitRange` type, so there should be no ambiguity about `Into<u8>`. 🤔

Reply:
It was "Could not infer type `T`" for ops.rs and "Into int32 was not implemented" for range_proof.rs . I thought the same thing too